.github/workflows/weekly.yml

```diff
@@ -26,8 +26,18 @@ jobs:
     if: github.repository_owner == 'apache'
     strategy:
       matrix:
-        flink: [1.17-SNAPSHOT, 1.18-SNAPSHOT]
+        flink_branches: [{
+          flink: 1.17.1,
+          branch: main
+        }, {
+          flink: 1.18-SNAPSHOT,
+          branch: main
+        }, {
+          flink: 1.17.1,
+          branch: v3.0
```

Comment:
It should be fine to build `v3.0` against 1.18-SNAPSHOT as well (once we resolve https://issues.apache.org/jira/browse/FLINK-32453).

Reply:
I've already added this